lib/src/validator/leak_detection.dart

```diff
@@ -0,0 +1,799 @@
+        'to ignore this. See $_falseSecretsDocumentationLink'
+      ].join('\n'));
+    } else if (leaks.isNotEmpty) {
+      // If we have 3 leaks we return all leaks, but only include the message
+      // about how ignore them in the last warning.
+      final lastLeak = leaks.removeLast();
+      errors.addAll([
+        ...leaks.take(2).map((leak) => leak.toString()),
+        [
+          lastLeak.toString(),
+          'Add a git-ignore style pattern to `false_secrets` in `pubspec.yaml`',
+          'to ignore this. See $_falseSecretsDocumentationLink',
+        ].join('\n'),
+      ]);
+    }
+  }
+}
+
+final _filesToIgnore = [
+  // We're not going to grep around inside what we assume are binary blobs
+  // If these include API keys of some sort it's hopefully because the author
+  // intended it so.
+  RegExp(r'\.(dill|bin|apk|so|a|dll|dylib|svg|png|jpe?g|gif|pdf)$'),
+  // Ignore anything inside node_modules, it's probably leaked from somewhere
+  // else, it's unlikely to be part of this package.
+  RegExp(r'(?:^|/)node_modules/'),
+  // Let's allow keys in files with DUMMY in the filename.
+  RegExp(r'/[^/]*dummy[^/]$', caseSensitive: false),
+  // .flat is a binary intermediate file, see:
+  // https://developer.android.com/studio/command-line/aapt2
+  RegExp(r'\.flat$'),
+  // Ignore non-secret Firebase project configuration, see:
+  // https://firebase.google.com/docs/projects/learn-more#config-files-objects
+  // While sharing this in source code is discouraged it's not a secret!
+  RegExp(r'(?:^|/)google-services\.json$'),
+  RegExp(r'(?:^|/)GoogleService-Info\.plist$'),
+  // Ignore: README.md, if it contains a key, it's probably illustrative.
+  RegExp('^README.md\$'),
+];
+
+/// Instance of a match against a [LeakPattern].
+@sealed
+class LeakMatch {
+  final LeakPattern pattern;
+  final SourceSpan span;
+
+  LeakMatch(this.pattern, this.span);
+
+  @override
+  String toString() =>
+      span.message('Potential leak of ${pattern.kind} detected.');
+}
+
+/// Definition of a pattern for detecting accidentally leaked secrets.
+@visibleForTesting
+@sealed
+class LeakPattern {
+  /// Human readable name for the kind of secret this pattern matches.
+  final String kind;
+
+  /// Pattern that matches a secret of given [kind].
+  final RegExp _pattern;
+
+  /// List of allow-listed patterns that are always known to be false-positives.
+  ///
+  /// Examples include dummy values commonly used for in documentation for
+  /// illustrative purposes.
+  final List<Pattern> _allowed;
+
+  /// List of allow-listed patterns, where the secret matched is ignored if the
+  /// 10 lines preceeding matches one of the [_allowedPreceeding] patterns.
+  ///
+  /// Examples include: `<meta name="google-signin-client_id" content="`
+  /// which clearly indicates that the `client_id` that might be considered
+  /// leaked is likely just used for a client-side sign-in flow.
+  final List<Pattern> _allowedPreceeding;
```

Comment:
Let's remove all these special cases... It's better than people just add files as `false_secrets`.